jurity/recommenders/intralist_diversity.py

```diff
@@ -0,0 +1,255 @@
+# -*- coding: utf-8 -*-
```

Comment:
Would it make sense to combine `intralist_diversity.py` and `interlist_diversity.py` into a single `diversity.py` module?

Reply:
that makes sense to me.

Reply:
And then there will be a one-to-one mapping between diversity.py and test_reco_diversity.py

Reply:
I think there is the opportunity to combine the two into a single module, for example, in ctr, there are `ips` and `dr` metrics. One thing that might be dangling for the two is they have some specific args, like intralist will need item_features. So would like to hear more comments.

Reply:
I think having them as two separate metrics (instead of CTR with 3 different methods) make more sense. `ips` and `dr` are different methods of calculating CTR, whereas interlist and intralist diversity fundamentally sounds like different metrics to me.

That being said, I think they can share the backend (ie an abstract class like `list_diversity`).

Reply:
@dorukkilitcioglu I think that's a good idea. so based on the discussions, the question is whether to 1. merge them into one class/one metric with different methods or 2. keep them separate but use a common base class, or 3. keep separate metrics but use one script. what do we think? @skadio @dorukkilitcioglu @takojunior @bkleyn @filip-michalsky

Reply:
I see where Doruk is coming from and I don't disagree with that. Still, I vote for Option 1 (merging them together).

The tricky bit with libraries, such as Jurity, that tries to do too many different things (such as fairness eval, bias mitigate, reco eval, classification, uplift perf, oh my..) is that the code comes very fragmented. 

As such, whenever there is a chance to modularize it and bundle components together, it helps the overall library to be more succinct. Xin's CTR example is a good one. Likewise, are you interested in diversity metrics for recommenders? Go check jurity/reco/diversity.py and see jurity/test/test_reco_diversity.py. 

Once combined under diversity.py, the implementation can be completely independent (option 2 above) OR using a base (option 3 above). 

On that choice, I am indifferent with slightly leaning toward Option 2 for now. I am not aware of any more upcoming diversity metric implementations that can benefit from extracting such a base class. This can be done in the future, **IF**, we need one more implementation and find ourselves repeating the code again --I doubt this case will happen any time soon. So I will leave that to Xin/Du on how they want it to be implemented at the moment.

Reply:
Totally agreed on that very last point - this is already a great contribution and it will be great to finally have it in Jurity regardless of the design choices on the backend (and I'm also happy to leave it to the authors of the diversity metric implementations).

@skadio Just to confirm your view, between the two options below, would you pick B?
```python
# Option A
m1 = DiversityRecoMetrics.InterListDiversity(...)
m2 = DiversityRecoMetrics.IntraListDiversity(...)

# Option B
m1 = DiversityRecoMetrics.ListDiversity(method='interlist', ...)
m2 = DiversityRecoMetrics.ListDiversity(method='intralist', ...)
```

Reply:
It is possible that I missed something here. 

I thought we all agree on Option A, the merge comment was only about pushing the code into a single diversity.py file. 

If we can’t get Option A —the discard my comments :)

Reply:
Pushing both classes to a single file makes sense!